File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(m, computers, network):
    stk = []
    stk.append(m)
    n = len(computers)
    while stk:
        k = stk.pop()
        for i in range(n):
            if k == i:
                continue
            if computers[k][i] == 1 and network[i] == 0:
                stk.append(i)
                network[i] = 2


def solution(n, computers):
    answer=0
    network = [0 for _ in range(n)]
    for i in range(n):
        if network[i] == 0:
            network[i] =1
            dfs(i,computers,network)
    return network.count(1)

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public boolean[] dfs(int[][] computers, boolean[] network, int i){

        for(int j=0;j<computers.length;j++){
            if(computers[i][j] == 1 && network[j] == false){
                network[j] = true;
                network = dfs(computers, network, j);
            }
        }
        
        return network;
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] network = new boolean[n];
        
        for(int i=0;i<computers.length;i++){
            if(network[i] == false){
                answer++;
                network = dfs(computers, network, i);
            }
        }
        
        
        
        return answer;
    }
}